[#66982] Remove jQuery UJS

oliverguenther opened the pull request:
https://community.openproject.org/wp/66982

It results in failures locally with turbo and the new build system, causing weird behaviors that I cannot reproduce on QA. This is a quick test of what would be needed

Relevant changes:
- [x] Replace `data-confirm: "Text"` with `data-turbo-confirm: "Text"`
- [x] Replace `data-method: :method` with `data-turbo-method: :method`
  - [x] Update affected controllers to use HTTP 303 status instead of 302 to ensure redirect requests are performed with a GET request

oliverguenther committed:
Replace confirm with turbo-confirm, method with turbo-method

oliverguenther committed:
Add disable-when-clicked controller when not in turbo

oliverguenther committed:
Remove ujs

NobodysNightmare commented on app/views/account/_password_login_form.html.erb:
```diff
@@ -64,7 +64,10 @@ See COPYRIGHT and LICENSE files for more details.
-                   data: { turbo_submits_with:  t(:label_loading) } %>
+                   data: {
+                     controller: "disable-when-clicked",
+                     "disable-when-clicked-text-value": t(:label_loading)
```
Just so that I understand this commit correctly: Inside turbo we still have the behaviour that buttons get disabled while their network request is pending and re-enabled, once it's done, correct? So this is only necessary here, because the login is not in fact using Turbo?

oliverguenther replied:
yes, correct 👍

NobodysNightmare commented:
Broad concern: While the changes _look_ good to me (as in: I understand what you are doing and why you are doing it), this PR still leaves me with the uneasy feeling that one of those 120 changes was done in error and it might be hard to figure out, if that piece is not covered with an automated test.

I guess QA is aware that we did changes that could crop up everywhere/in unexpected places?

HDinger committed:
Add `see_other` status to categories controller

oliverguenther committed:
Fix routing error on css maps

Falls back to a 404 if not found by esbuild

oliverguenther committed:
Fix render of categories

oliverguenther committed:
Wait on filter after page load

For some reason, the synchronization for cuprite doesn't seem to be
correct after page load in all cases. A retry_block is all it takes

oliverguenther committed:
Fix method on project index

myabc commented on app/components/enumerations/row_component.rb:
```diff
@@ -64,8 +64,8 @@ def delete_link
         helpers.enumeration_path(enumeration),
-        method: :delete,
-        data: { confirm: I18n.t(:text_are_you_sure) },
+        turbo_method: :delete,
+        data: { turbo_confirm: I18n.t(:text_are_you_sure) },
```
```suggestion
        data: { turbo_method: :delete, turbo_confirm: I18n.t(:text_are_you_sure) },
```

myabc commented on app/components/projects/settings/index_page_header_component.html.erb:
```diff
@@ -81,7 +81,7 @@
           href: project_archive_path(@project, status: "", name: @project.name),
           content_arguments: {
-            data: { confirm: t("project.archive.are_you_sure", name: @project.name), method: :post },
+            data: { turbo_confirm: t("project.archive.are_you_sure", name: @project.name), method: :post },
```
```suggestion
            data: { turbo_method: :post, turbo_confirm: t("project.archive.are_you_sure", name: @project.name) },
```

myabc commented on app/views/forums/index.html.erb:
```diff
@@ -153,8 +153,8 @@ See COPYRIGHT and LICENSE files for more details.
-                                            data: { confirm: t(:text_are_you_sure) },
-                                            method: :delete,
+                                            data: { turbo_confirm: t(:text_are_you_sure) },
+                                            turbo_method: :delete,
```
```suggestion
                                            data: { turbo_method: :delete, turbo_confirm: t(:text_are_you_sure) },
```

myabc commented on app/views/groups/_memberships.html.erb:
```diff
@@ -42,75 +42,78 @@ See COPYRIGHT and LICENSE files for more details.
                       </span>
-                    </div>
                   </div>
-                </th>
-                <th>
-                  <div class="generic-table--sort-header-outer">
-                    <div class="generic-table--sort-header">
+                </div>
+              </th>
+              <th>
+                <div class="generic-table--sort-header-outer">
+                  <div class="generic-table--sort-header">
                       <span>
                         <%= t(:label_role_plural) %>
                       </span>
-                    </div>
                   </div>
-                </th>
-                <th><div class="generic-table--empty-header"></div></th>
-              </tr>
+                </div>
+              </th>
+              <th>
+                <div class="generic-table--empty-header"></div>
+              </th>
+            </tr>
```
nitpick: indentation is wrong.

myabc commented on app/views/messages/show.html.erb:
```diff
@@ -66,8 +66,8 @@ See COPYRIGHT and LICENSE files for more details.
-                      method: :delete,
-                      data: { confirm: t(:text_are_you_sure) },
+                      turbo_method: :delete,
+                      data: { turbo_confirm: t(:text_are_you_sure) },
```
```suggestion
                      data: { turbo_method: :delete, turbo_confirm: t(:text_are_you_sure) },
```

myabc commented on app/views/my/access_token_partials/_storage_tokens_section.html.erb:
```diff
@@ -63,9 +63,9 @@ See COPYRIGHT and LICENSE files for more details.
+                              data: { turbo_confirm: t("my_account.access_tokens.storages.revoke_token", storage: token.oauth_client.integration.name),
                                       test_selector: "storages-token-row-#{token.id}-revoke" },
-                              method: :delete,
+                              turbo_method: :delete,
```
```suggestion
                              data: {
                                turbo_method: :delete,
                                turbo_confirm: t("my_account.access_tokens.storages.revoke_token", storage: token.oauth_client.integration.name),
                                test_selector: "storages-token-row-#{token.id}-revoke"
                              },
```

myabc commented on app/views/news/show.html.erb:
```diff
@@ -99,8 +100,8 @@ See COPYRIGHT and LICENSE files for more details.
-                                    data: { confirm: t(:text_are_you_sure) },
-                                    method: :delete,
+                                    data: { turbo_confirm: t(:text_are_you_sure) },
+                                    turbo_method: :delete,
```
```suggestion
                                    data: { turbo_method: :delete, turbo_confirm: t(:text_are_you_sure) },
```

myabc commented on app/views/repositories/destroy_info.html.erb:
```diff
@@ -60,7 +60,7 @@ See COPYRIGHT and LICENSE files for more details.
       <div class="danger-zone--verification">
         <input type="text">
         <%= styled_button_tag project_repository_path(@project),
-                              method: :delete,
+                              turbo_method: :delete,
                               title: t(:button_delete),
                               disabled: true,
                               class: "-primary" do %>
```
I don't think Turbo does anything special with buttons. 🤔

myabc commented on config/routes.rb:
```diff
@@ -55,7 +55,10 @@
 
   # Respond with 410 gone for APIV2 calls
   match "/api/v2(/*unmatched_route)", to: proc { [410, {}, [""]] }, via: :all
+
+  # Respond with 404 with common assets
   match "/assets/compiler.js.map", to: proc { [404, {}, [""]] }, via: :all
+  match "*.css.map", to: proc { [404, {}, [""]] }, via: :all
```
This doesn't appear related - I'm guessing there's an issue with sourcemaps in the new set up?

oliverguenther replied:
I added this as it became annoying to test headful, causing routing errors on some missing source map

myabc commented on modules/overviews/app/views/overviews/overviews/show.html.erb:
```diff
@@ -45,7 +45,7 @@
           href: project_archive_path(@project, status: "", name: @project.name),
           content_arguments: {
-            data: { method: :post, turbo: false, confirm: t("project.archive.are_you_sure", name: @project.name) }
+            data: { turbo_method: :post, turbo: false, turbo_confirm: t("project.archive.are_you_sure", name: @project.name) }
```
This doesn't make sense.

`turbo_method: :post, turbo: false`

myabc replied:
I've changed to a Turbo request, *but* the redirect does not work as expected.

myabc commented on modules/reporting/lib/widget/controls/delete.rb:
```diff
@@ -41,7 +41,7 @@ def render
       opt1 = link_to I18n.t(:button_delete),
                      url_for(url_opts),
-                     method: :delete,
+                     turbo_method: :delete,
```
```suggestion
                     data: { turbo_method: :delete },
```

myabc commented on modules/reporting/lib/widget/table/entry_table.rb:
```diff
@@ -175,8 +175,8 @@ def icons(result)
-                           data: { confirm: I18n.t(:text_are_you_sure) },
-                           method: :delete,
+                           data: { turbo_confirm: I18n.t(:text_are_you_sure) },
+                           turbo_method: :delete,
```
```suggestion
                           data: { turbo_method: :delete, turbo_confirm: I18n.t(:text_are_you_sure) },
```

myabc reviewed (changes requested):
I've found a few mistakes, as well as things I'm unsure about. Please see inline comments!

myabc committed:
Update more actions to respond with 303 See Other

oliverguenther committed:
use data-turbo-method in destroy_info of repositories

myabc committed:
Rework Reporting controller using Request.JS

oliverguenther committed:
Fix deletion of messages and news

oliverguenther committed:
Adapt session spec to use cuprite

myabc reviewed (approved):
🚢
